packages/flame/lib/src/game/camera/camera_component.dart

```diff
@@ -0,0 +1,43 @@
+import 'dart:ui';
+import '../../../components.dart';
+import 'camera.dart';
+
+class CameraComponent extends Component {
+  CameraComponent(this.camera, this.world);
+
+  final Camera camera;
+  final ComponentSet world;
+
+  @override
+  void update(double dt) {
+    super.update(dt);
+    camera.update(dt);
+  }
+
+  @override
+  void render(Canvas canvas) {
+    // TODO(st-pasha): it would be easier to keep the world and the
+    //   HUD as two separate component trees.
+    camera.viewport.render(canvas, (_canvas) {
+      // First render regular world objects
+      canvas.save();
+      camera.apply(canvas);
+      world.forEach((component) {
+        if (!component.isHud) {
+          canvas.save();
+          component.renderTree(canvas);
+          canvas.restore();
+        }
+      });
+      canvas.restore();
+      // Then render the HUD
+      world.forEach((component) {
```

Comment:
I wonder the perfomance impacts of running through the component set twice

Reply:
Agree, running through the list twice is suboptimal. Ideally, we would want to have them as two separate component trees: one is the **world**, and another is the **hud**, and then the performance concerns would disappear. However, I left this refactoring step for the future, since it requires #906 to be merged.

In the meanwhile, it is not clear whether this double-loop setup made things better or worse. On the one hand, as you noted, we iterate through the component tree twice. On the other hand, we no longer need to apply camera transform for every non-hud component: this is done only once outside the loop. Plus, the new approach ensures that any hud components are actually drawn on top of the world, not below it.